packages/services/api/src/modules/schema/providers/schema-publisher.ts

```diff
@@ -1040,6 +1040,23 @@ export class SchemaPublisher {
       this.schemaManager.getMaybeLatestVersion(target),
     ]);
 
+    if (!latestVersion && input.service) {
```

Comment:
I think we would rather have to check whether the latestVersion has a service that is named like this service? 🤔

Reply:
There would also be an edge case, where someone deletes a service (`hive schema:delete`) and tries to republish it, but then it fails, because it would no longer be in the previous version 🤔

Reply:
Youre right about the latestVersion. I'll fix that.

For the edge case, I think we can ignore that. If they are deleting then we can treat like new.

Reply:
Added validation to schema check also. This will prevent us from having issues where the check passes but push fails because of the service name.